modules/tide_media/tide_media.module

```diff
@@ -464,3 +466,35 @@ function tide_media_file_presave(FileInterface $file) {
     $file->setFilename($filename_new);
   }
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function tide_media_preprocess_page(&$variables) {
+  $route_name = \Drupal::routeMatch()->getRouteName();
+  $current_user = \Drupal::currentUser();
+  $roles = $current_user->getRoles();
+
+  if ($route_name === 'entity.user.edit_form' && in_array('secure_file_user', $roles)) {
+    $url = Url::fromRoute('entity.user.canonical', ['user' => $current_user->id()]);
+    $response = new RedirectResponse($url->toString());
+    $response->send();
+    exit();
+  }
+}
+
+/**
+ * Implements hook_menu_local_tasks_alter().
+ */
+function tide_media_menu_local_tasks_alter(&$data, $route_name, RefinableCacheableDependencyInterface &$cacheability) {
+  $current_user = \Drupal::currentUser();
+  $roles = $current_user->getRoles();
+
+  if (in_array('secure_file_user', $roles)) {
+    foreach ($data['tabs'][0] as $key => $items) {
```

Comment:
Iterating over all $data['tabs'][0] entries can cause a notice or warning if 'tabs' or its children are not structured as expected.
Should change this to check everything is set and cover the validation check.
Something like this may be - 
```
if (in_array('secure_file_user', $roles, TRUE) && isset($data['tabs'])) {
    foreach ($data['tabs'] as $key => &$tabs) {
      if (is_array($tabs)) {
        foreach ($tabs as $tab_key => $tab) {
          if ($tab_key === 'entity.user.edit_form') {
            unset($tabs[$tab_key]);
          }
        }
      }
    }
  }
```

Please check if this works in this scenario

Reply:
@MdNadimHossain , you are right. I worked in the ideal scenario of tabs/children are structured. Thank you for the code.